support/jakarta-ee/src/main/java/org/apache/shiro/ee/cdi/AnnotatedTypeWrapper.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.shiro.ee.cdi;
+
+import java.lang.annotation.Annotation;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.enterprise.inject.spi.AnnotatedType;
+import lombok.Getter;
+import lombok.experimental.Delegate;
+
+/**
+ * Wraps annotation types to facilitate additional annotations for CDI
+ *
+ * @author lprimak
+ * @param <T> type of annotated class
+ */
+class AnnotatedTypeWrapper<T> implements AnnotatedType<T> {
+    // the below is so the compiler doesn't complain about unchecked casts
+    private abstract class AT implements AnnotatedType<T> { }
+    private final @Delegate(types = AT.class) AnnotatedType<T> wrapped;
+    private final @Getter Set<Annotation> annotations;
```

Comment:
This doesn't justify lombok, we can use our IDEs here.

Reply:
IDE won't maintain the generated code. Neither does 'delombok'
Lombok discussion in slack